tools/traffic/workers/blob_writer.go

```diff
@@ -0,0 +1,164 @@
+	writeSuccessMetric metrics.CountMetric
+
+	// writeFailureMetric is used to record the number of failed write requests.
+	writeFailureMetric metrics.CountMetric
+}
+
+// NewBlobWriter creates a new BlobWriter instance.
+func NewBlobWriter(
+	ctx *context.Context,
+	waitGroup *sync.WaitGroup,
+	logger logging.Logger,
+	config *config2.WorkerConfig,
+	disperser clients.DisperserClient,
+	unconfirmedKeyHandler KeyHandler,
+	generatorMetrics metrics.Metrics) BlobWriter {
+
+	var fixedRandomData []byte
+	if config.RandomizeBlobs {
+		// New random data will be generated for each blob.
+		fixedRandomData = nil
+	} else {
+		// Use this random data for each blob.
+		fixedRandomData = make([]byte, config.DataSize)
+		_, err := rand.Read(fixedRandomData)
+		if err != nil {
+			panic(fmt.Sprintf("unable to read random data: %s", err))
+		}
+		fixedRandomData = codec.ConvertByPaddingEmptyByte(fixedRandomData)
+	}
+
+	return BlobWriter{
+		ctx:                   ctx,
+		waitGroup:             waitGroup,
+		logger:                logger,
+		config:                config,
+		disperser:             disperser,
+		unconfirmedKeyHandler: unconfirmedKeyHandler,
+		fixedRandomData:       &fixedRandomData,
+		writeLatencyMetric:    generatorMetrics.NewLatencyMetric("write"),
+		writeSuccessMetric:    generatorMetrics.NewCountMetric("write_success"),
+		writeFailureMetric:    generatorMetrics.NewCountMetric("write_failure"),
+	}
+}
+
+// Start begins the blob writer goroutine.
+func (writer *BlobWriter) Start() {
+	writer.waitGroup.Add(1)
+	ticker := time.NewTicker(writer.config.WriteRequestInterval)
+
+	go func() {
+		defer writer.waitGroup.Done()
+
+		for {
+			select {
+			case <-(*writer.ctx).Done():
+				return
+			case <-ticker.C:
+				writer.writeNextBlob()
+			}
+		}
+	}()
+}
+
+// writeNextBlob attempts to send a random blob to the disperser.
+func (writer *BlobWriter) writeNextBlob() {
+	data := writer.getRandomData()
+	key, err := metrics.InvokeAndReportLatency(writer.writeLatencyMetric, func() ([]byte, error) {
+		return writer.sendRequest(*data)
+	})
+	if err != nil {
+		writer.writeFailureMetric.Increment()
+		writer.logger.Error("failed to send blob request", "err:", err)
+		return
+	}
+
+	writer.writeSuccessMetric.Increment()
+
+	checksum := md5.Sum(*data)
+	writer.unconfirmedKeyHandler.AddUnconfirmedKey(key, checksum, uint(len(*data)))
+}
+
+// getRandomData returns a slice of random data to be used for a blob.
+func (writer *BlobWriter) getRandomData() *[]byte {
+	if *writer.fixedRandomData != nil {
+		return writer.fixedRandomData
+	}
+
+	data := make([]byte, writer.config.DataSize)
+	_, err := rand.Read(data)
+	if err != nil {
+		panic(fmt.Sprintf("unable to read random data: %s", err))
+	}
+	data = codec.ConvertByPaddingEmptyByte(data)
+
+	return &data
+}
+
+// sendRequest sends a blob to a disperser.
+func (writer *BlobWriter) sendRequest(data []byte) ([]byte /* key */, error) {
```

Comment:
you could use a named return values if the returns values look ambiguous
`...sendRequest(data []byte) (key []byte, err error)`

Reply:
Interesting, today I learned that syntax exists. :) Change made.